### DISCUSSION TITLE
ListView Selection does not work in Android

### DISCUSSION THREAD

**Discussion by markinnh** [unoplatform/uno#9076] [Q&A] [answered]
I have a test app and I have been unable to get ListView Selection to work in Android.  I am adding the ListView Xaml, maybe there is something missing for Android that I don't know about.  I am running the latest release in Visual Studio 2022 17.4 Preview 2 and recently installed the Uno Templates (6/16), I have not installed any other extensions.
```xaml
<ListView Name="BrowseInv" Grid.Row="2"  ItemsSource="{x:Bind CollectionView.View}" ItemTemplate="{StaticResource DisplayItem}" 
                  SelectionChanged="BrowseInv_Changed" VerticalAlignment="Stretch" 
                              Width="230"
                              SelectedItem="{x:Bind SelectedItem}">
                        <ListView.ItemsPanel>
                            <ItemsPanelTemplate>
                                <ItemsWrapGrid MaximumRowsOrColumns="2" Orientation="Horizontal"/>
                            </ItemsPanelTemplate>
                        </ListView.ItemsPanel>
                        <ListView.GroupStyle>
                            <GroupStyle HeaderTemplate="{StaticResource DisplayHeader}"/>
                        </ListView.GroupStyle>
                    </ListView>
```

**Comment by jeromelaban** [comment]
Thanks for the report. 

It's likely that the issue you're seeing is related to the use of `x:Bind` and its default mode of `OneTime`. You can try with `SelectedItem="{x:Bind SelectedItem, Mode=TwoWay}"` and see if it helps.

Otherwise, you can try to see if this works on UWP or WinUI and let us know.

**Reply by markinnh** [reply]
It does work on WinUI3, no problem, but I have the SelectionChanged event hooked to a handler and nothing happens in Android.  Also, I set up a project with the same XAML in UWP and it works in UWP but not in Android.

**Reply by jeromelaban** [reply]
Thanks. Could you create a repro project which shows what you're experiencing?

**Comment by markinnh** [comment]
This is the shared application, the entire project zipped is 200 Mb, so I can't do that.  All my code (XAML/C#) is in these files.  It does work in UWP/WinUI3, but fails when I try to run it on Android.

Thanks,
Mark

________________________________

**Reply by jeromelaban** [reply]
If you want to zip the project, you can remove all the bin and obj folders.

**Reply by markinnh** [reply]
I removed all the bin and obj files, the project is zipped
[UnoTestApp2.zip](https://github.com/unoplatform/uno/files/8936977/UnoTestApp2.zip)
.

**Comment by markinnh** [comment, marked as answer]
Found the problem, can't use CollectionViewSource.

**Reply by jeromelaban** [reply]
Thanks for the update. It's likely that we have an issue in `CollectionViewSource`, it's supposed to be supported properly.

**Reply by jeromelaban** [reply]
The issue you've faced here is https://github.com/unoplatform/uno/issues/7494, where it's generally best to avoid providing a lazy enumerable to collections for performance reasons, even on WinUI/UWP.

If you use `.ToArray()` at the end, the behavior is the right one:
```csharp
        public IEnumerable<IGrouping<int, Item>> GroupedItems => (from item in GetItems()
                                                                 group item by item.GroupId into g
                                                                 orderby g.Key
                                                                 select g).ToArray();

```

In all cases, it's best to expose an `ObservableCollection<>` instead, as it is built to handle updates properly, instead of relying on eventual binding refresh.
